### DISCUSSION TITLE
Is the RoadRunner - broadcast plugin compatible with Pusher or Socket.IO

### DISCUSSION THREAD

**Discussion by litan1106** [roadrunner-server/roadrunner#1028] [Q&A] [answered]
Is the RoadRunner - broadcast plugin compatible with Pusher or Socket.IO

https://laravel.com/docs/9.x/broadcasting#pusher-compatible-open-source-alternatives

**Comment by rustatian** [comment, marked as answer]
Hey @litan1106. No, the broadcast plugin hasn't implemented the `Laravel Echo` API. But, if you make an issue with the detailed Laravel Echo API description, it could be implemented as one of the broadcast drivers.
In general, the broadcast plugin accepts different drivers (Redis, boltdb, memory, etc). The driver connects to the broadcast plugin to provide broadcast capabilities. For the Laravel Echo, that could be a similar driver.

**Reply by litan1106** [reply]
Is the RoadRunner - broadcast plugin compatible with Pusher or Socket.IO?

Either of them will make things work nicely with existing frameworks.

Example:
soketi is nodejs based websocket server implements the pusher protocol. https://docs.soketi.app

**Reply by rustatian** [reply]
> Is the RoadRunner - broadcast plugin compatible with Pusher or Socket.IO?

@litan1106 
No, it's not compatible, but Broadcast is just a collector of the drivers. Basically, you can implement the [`pubsub` API interface](https://github.com/roadrunner-server/api/blob/master/plugins/pubsub/interface.go) and support any third-party framework as a broadcast driver.
